src/backend/core/models.py

```diff
@@ -284,8 +284,27 @@ def __str__(self):
         return f"{id_str:s}{main_str:s}"
 
     def save(self, *args, **kwargs):
-        """Ensure users always have one and only one main identity."""
+        """
+        Save identity, ensuring user always have one and only one main identity.
+        Also convert valid invitations if new identity.
```

Comment:
might be cool to explain to what are converted the valid invitation !

Reply:
it's explained just below but I'll add it to the docstrings too.